src/chocolatey.tests.integration/scenarios/UpgradeScenarios.cs

```diff
@@ -3525,5 +3525,105 @@ public void should_not_have_executed_beforemodify_hook_script_for_upgrade_versio
                 MockLogger.contains_message("pre-beforemodify-all.ps1 hook ran for upgradepackage 1.1.0", LogLevel.Info).ShouldBeFalse();
             }
         }
+
+        public class when_upgrading_a_package_with_beforeModify_script_with_dependencies_with_beforeModify_scripts : ScenariosBase
+        {
+            private const string TargetPackageName = "hasdependencywithbeforemodify";
+            private const string DependencyName = "isdependencywithbeforemodify";
+
+            public override void Context()
+            {
+                base.Context();
+
+                Scenario.add_packages_to_source_location(Configuration, "{0}.*".format_with(TargetPackageName) + Constants.PackageExtension);
+                Scenario.add_packages_to_source_location(Configuration, "{0}.*".format_with(DependencyName) + Constants.PackageExtension);
+                Scenario.install_package(Configuration, DependencyName, "1.0.0");
+                Scenario.install_package(Configuration, TargetPackageName, "1.0.0");
+
+                Configuration.PackageNames = Configuration.Input = TargetPackageName;
+            }
+
+            public override void Because()
+            {
+                Results = Service.upgrade_run(Configuration);
+            }
+
+            [Fact]
+            public void should_upgrade_the_package()
+            {
+                var packageFile = Path.Combine(Scenario.get_top_level(), "lib", TargetPackageName, "{0}.nupkg".format_with(TargetPackageName));
+                var package = new OptimizedZipPackage(packageFile);
+                package.Version.Version.to_string().ShouldEqual("2.0.0.0");

```

Comment:
Why is this a four part version number?

Reply:
`Version.Version` is a `System.Version` and will give a four-part version. I'm unsure exactly why this route was chosen, I simply copied the existing patterns here to avoid reinventing the wheel ^^.